### ISSUE TITLE
Please support changing recovery dir, or disabling recovery

### ISSUE THREAD

**Issue by joshtriplett** [nfs-ganesha/nfs-ganesha#543]
If Ganesha can't access `/var/lib/nfs`, it emits a large number of error messages regarding failed attempts to create recovery files underneath that path.

I'd like to be able to either disable recovery (since I'll be running Ganesha with an FSAL for purposes that don't require the ability to recover) or change the recovery path via configuration (rather than hard-coding it to `@SYSSTATEDIR@/lib/nfs/ganesha`).

This would also make it more feasible for multiple unrelated parallel instances of ganesha on the same system to coexist without interfering with each other.

**Comment by dang** [comment]
You can compile Ganesha to install in a prefix.  This is how I do my development and testing.  It will then put everything in <prefix>/etc <prefix>/usr <prefix>/var, etc.  Beyond that, code changes would be necessary.

**Comment by joshtriplett** [comment]
@dang That's what I ended up doing, but it would be nice to have runtime configurability of that, which would allow running multiple instances in parallel from the same binary.

**Comment by ffilz** [comment]
Feel free to submit a patch for this. I'm not sure if it should be a command line option or a config file option, one advantage of being in the config file is that the fact that the config file is intended to work with a specific instance of Ganesha and thus have a specific location of the recovery directory would be tied together. On the other hand, we have log file specification on the command line in addition to in the config file, so maybe both places.

**Comment by ffilz** [comment]
Leaving this for backlog, it does seem a useful and small feature request.

**Comment by ffilz** [comment]
See patch https://review.gerrithub.io/c/ffilz/nfs-ganesha/+/514184
